mlibrary/prejudice
# [SEARCH-1223] Implement idle detection in Prejudice

# SEARCH-1223
_This PR resolves [SEARCH-1223](https://tools.lib.umich.edu/jira/browse/SEARCH-1223)._
## Overview
Prejudice updates information for Favorites and affiliated data every few minutes, even when idle. This PR adds idle detection to prevent that from happening.
### IdleTimeout
The singleton `IdleTimeout.js` was created. [idleTimeout](https://github.com/jackmu95/idle-timeout/) was installed and imported for the idle detection.
## What does this PR affect?
This PR affects `Profiles.js`, `FavoritesList.js`, and `FavoritesSuggest.js`.
## Testing
* In this repository, open `src/Prejudice/IdleTimeout.js` and add `console.log('Idle detected.');` to the callback in `line 11`.
* `npm run build`
* Go to the [Search](https://github.com/mlibrary/search) repository and run `npm install`.
* Overwrite `node_modules/prejudice/` with the Prejudice repository.
* `npm run start`
* Go to `http://localhost:3000/everything` and wait to see if `Idle detected.` shows up in the console logs.
